doc/sphinx-guides/source/admin/mail-groups.rst

```diff
@@ -55,6 +55,36 @@ To load it into your Dataverse installation, either use a ``POST`` or ``PUT`` re
 
 ``curl -X POST -H 'Content-type: application/json' http://localhost:8080/api/admin/groups/domain --upload-file domainGroup1.json``
 
+Matching with domains or regular expressions
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Adding simple domain names requires exact matches of user email domains and the configured domains of a group.
+Although you could add multiple domains to a group, those still require exact matches.
+
+In addition, you can use one or multiple regular expressions instead of simple domains for a group. Those
+should not be mixed, although it would work. Regular expressions still require exact matches, but are way more flexible.
+
+Some hints:
+
+- Due to their excessive CPU usage, regular expressions should be used rarely.
+- Also due to their CPU usage, all domain groups are cached. Cache updates when groups change via API.
```

Comment:
The singleton has a copy of the groups in memory (instance variables).

Everytime authorization happens, the regular expressions aren't recompiled, which would be a huge waste of CPU resources. In other words: the singleton is a facade for the rarely changing group data.

This "cache" is updated (compiling the regex) on startup and when domain groups are changed via C~~R~~UD API actions (not during read, obviously).

This might not be necessary to mention in the guides, as an admin should never notice anything of it. I left it there for FWIW.

Reply:
Ah, yeah, I think Admins won't care about that detail - I think it's safe to remove.